src/components/Movies/Movies.js

```diff
@@ -1,12 +1,136 @@
 import React, { Component } from 'react';
+import './movies.scss';
+//import './Movies.scss';
 import Button from '../common/Button/Button';
+import MovieCard from './MovieCard/MovieCard';
+import axios from 'axios';
+import { URL_MOVIE, MOVIE_ON_PAGE } from '../../constants/api';
+import Loader from '../common/Loader/Loader';
+import Filters from "./Filters/Filters";
+
 
 class Movies extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            movies: [],
+            filters: {
+                title: '',
+                language: '',
+                status:  '',
+                release_date_first: '',
+                release_date_last: '',
+                budget_min: 0,
+                budget_max: 500000000,
+                popularity_min: 0,
+                popularity_max: 200,
+                revenue_min: 0,
+                revenue_max: 30000,
+            },
+            isFiltersSet: false,
+            currentPage: 1,
+            isFiltersHidden : true,
+            isLoading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.getMovies(this.getCurrentUrl(this.state.currentPage));
+    }
+
+    setBudget = (minBudget, maxBudget) => this.setState({filters : {...this.state.filters, budget_min: minBudget,budget_max: maxBudget}});
+    setPopularity = (minPopularity, maxPopularity) => this.setState({filters :{...this.state.filters, popularity_min: minPopularity, popularity_max: maxPopularity}});
+    setVote = (minVote, maxVote) => this.setState({filters :{...this.state.filters,revenue_min: minVote,revenue_max: maxVote}});
+    setLanguage = (language) => this.setState({filters :{...this.state.filters,language: language}});
+    setStatus = (status) => this.setState({filters :{...this.state.filters,status: status}});
+    setSearchTitle = (title) => this.setState({filters :{...this.state.filters,title: title}});
+    setMinDate = (date) => this.setState({filters :{...this.state.filters,release_date_first: date}});
+    setMaxDate = (date) => this.setState({filters :{...this.state.filters,release_date_last: date}});
+
+    toggleFilters = () =>  this.setState({isFiltersHidden: !this.state.isFiltersHidden});
+    setFilter = (isFiltersSet) => this.setState({isFiltersSet: isFiltersSet})
+
+    getUrl = () => {
+        let result = `${URL_MOVIE}?`;
+        Object.entries(this.state.filters).forEach((element) => {
+            if (element[1] !== '') {
+                result += `&${element[0]}=${element[1].toString().replace(/,/g, '').replace(/ /g, '%20')}`;
+            }
+        });
+        return result;
+    };
+
+    getCurrentUrl = (page) => {
+        return (this.state.isFiltersSet
```

Comment:
this.state.isFiltersSet - unnessary state value. You can simply check if initialFilters values are equal with current ones